### PR TITLE
Feature: http-error-handler to support error headers

### LINKED ISSUE #892: `http-error-handler` does not respect headers set for an HTTP Error
**Describe the bug**
`createError` function from the [http-errors](https://npm.im/http-errors) package allows setting HTTP headers to be sent to the client. `http-error-handler` however, ignores those.

**To Reproduce**
How to reproduce the behaviour:
```ts
async function getTheThing() {
  throw createError(404, { message: 'Failed to get the thing', headers: { foo: 'bar' } });
}

async function theThingHandler(event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> {
  const theThing = await getTheThing();

  return {
    statusCode: StatusCodes.HTTP_STATUS_OK,
    body: JSON.stringify(theThing),
  };
}

const handler = middy(theThingHandler).use(httpErrorHandler());
```
**Current behaviour**

```JSON
{"Content-Type": "application/json"}
```

**Expected behaviour**
```JSON
{"Content-Type": "application/json", "foo": "bar"}
```

**Environment (please complete the following information):**
 - Node.js: 14.18.2
 - Middy: 3.2.2
 - AWS SDK: 2.1097.0

### PR DESCRIPTION
**Pull request by willfarrell**
@squ1b3r

Closes #892

### CODE REVIEW

File: packages/http-error-handler/__tests__/index.js [hunk 1]
```diff
@@ -99,6 +99,29 @@ test('It should be possible to pass a custom logger function', async (t) => {
   t.true(logger.calledWith(expectedError))
 })
 
+test('It should be possible to pass in headers with error', async (t) => {
+  const handler = middy(() => {
+    const error = createError(422, 'Unprocessable Entity')
+    error.headers = {
```
Comment (squ1b3r) [inline packages/http-error-handler/__tests__/index.js:105]: You can also pass headers to the `createError` like this:
```ts
createError(424, { message: 'Unprocessable Entity', headers: { Location: 'http://exmaple.org/500' } });
```
but this also works.

**Review by squ1b3r — APPROVED** [review]
LGTM! Thanks!

File: packages/http-error-handler/__tests__/index.js (hunk 1 above)
Comment (willfarrell) [reply packages/http-error-handler/__tests__/index.js:105]: The `createError` function here is a slimmed down version built into middy. It only has the functionality needed for the middlewares.

**Review by squ1b3r — APPROVED** [review]

**Comment by squ1b3r** [comment]
@willfarrell once again, thank you for fixing this so quickly. This is going to save me a lot of time and pain. When do you think you can release this?

**Comment by willfarrell** [comment]
I'll try for later today. Before Monday for sure.

**Pull request merged**